lib/Graph/Graph.cpp

```diff
@@ -62,10 +62,13 @@ void Module::strip() {
 }
 
 void Module::clear() {
-  eraseFunctions();
-
   for (auto it = constants_.begin(), e = constants_.end(); it != e; it++) {
     Constant *v = *it;
+
+    for (auto F : functions_) {
+      F->getLogContext().logNodeDeletion(*v);
```

Comment:
What if `F` doesn't use `v`? This seems incorrect to me since we would have logged only deleting it but not adding it.

It feels to me like Constant modifications should be logged at the Module level. Even if we modify or add a Constant during a Function optimization, it should probably be logged somehow that the Module's Constant was modified/added, but noting that it was during optimizing `F`.

Reply:
You are right, it will create confusion.  Maybe it's better to specify in the log that the constant is deleted from module instead of certain function.

I'm thinking of adding another logging method 'logConstantOrPlaceholderDeletion'. In this method, the log will be produced as "DELETE Constant XXX from Module XYZ", so that later when we processing the log the parser knows that the deleted constant may be used in another function.